src/skvhttpclient/src/skvhttp/client/SKVClient.cpp

```diff
@@ -0,0 +1,157 @@
+
+boost::future<Response<>> Client::createCollection(dto::CollectionMetadata metadata, std::vector<String> rangeEnds) {
+  dto::CollectionCreateRequest request {
+    .metadata = std::move(metadata),
+    .rangeEnds = std::move(rangeEnds)
+  };
+  
+  return _HTTPClient.POST<dto::CollectionCreateRequest>("/api/CreateCollection", std::move(request));
+}
+
+boost::future<Response<>> Client::createSchema(const String& collectionName, const dto::Schema& schema) {
+  dto::CreateSchemaRequest request {
+    .collectionName = collectionName,
+    .schema = schema
+  };
+
+  return _HTTPClient.POST<dto::CreateSchemaRequest>("/api/CreateSchema", std::move(request));
+}
+
+boost::future<Response<std::shared_ptr<dto::Schema>>> Client::getSchema(const String& collectionName, const String& schemaName, int64_t schemaVersion) {
+  SchemaCacheT::const_iterator collectionIt = _schemaCache.find(collectionName);
+  if (collectionIt != _schemaCache.end()) {
+    auto schemaNameIt = collectionIt->second.find(schemaName);
+    if (schemaNameIt != collectionIt->second.end()) {
+      if (schemaVersion == dto::ANY_SCHEMA_VERSION && schemaNameIt->second.size()) {
+        std::shared_ptr<dto::Schema> schema = schemaNameIt->second.begin()->second;
+        return MakeResponse<std::shared_ptr<dto::Schema>>(Statuses::S200_OK("Got schema from cache"), std::move(schema));
+      }
+      
+      auto schemaVersionIt = schemaNameIt->second.find(schemaVersion);
+      if (schemaVersionIt != schemaNameIt->second.end()) {
+        std::shared_ptr<dto::Schema> schema = schemaVersionIt->second;
+        return MakeResponse<std::shared_ptr<dto::Schema>>(Statuses::S200_OK("Got schema from cache"), std::move(schema));
+      }
+    }
+  }
+  
+  dto::GetSchemaRequest request {
+    .collectionName = collectionName,
+    .schemaName = schemaName,
+    .schemaVersion = schemaVersion
+  };
+
+  return _HTTPClient.POST<dto::GetSchemaRequest, dto::GetSchemaResponse>("/api/GetSchema", std::move(request)).
+    then([this, collectionName, schemaName, schemaVersion] (auto&& futResp) {
+      auto&& [status, resp] = futResp.get();
+      auto schema = std::make_shared<dto::Schema>(resp.schema);
+      if (status.is2xxOK()) {
+        _schemaCache[collectionName][schemaName][schemaVersion] = schema;
+      }
+      return Response<std::shared_ptr<dto::Schema>>(std::move(status), schema);
+    });
+}
+
+boost::future<Response<TxnHandle>> Client::beginTxn(dto::TxnOptions options) {
+  dto::TxnBeginRequest request {
+    .options = std::move(options)
+  };
+
+   return _HTTPClient.POST<dto::TxnBeginRequest, dto::TxnBeginResponse>("/api/TxnBegin", std::move(request)).
+    then([this] (auto&& futResp) {
+      auto&& [status, resp] = futResp.get();
+      TxnHandle txn(this, resp.timestamp);
+      return Response<TxnHandle>(std::move(status), std::move(txn));
+    });
+}
+
+boost::future<Response<>> TxnHandle::write(dto::SKVRecord& record, bool erase, dto::ExistencePrecondition precondition) {
```

Comment:
will a partial update be a separate API? if not, maybe just take the vector of fields as a param here

Reply:
yeah, the plan is to have partial update separate